chain/chain/src/chain.rs

```diff
@@ -2761,6 +2761,20 @@ impl Chain {
             );
             store_update.commit()?;
             flat_storage_manager.create_flat_storage_for_shard(shard_uid).unwrap();
+            // Flat storage is ready, load memtrie if it is enabled.
+            let memtrie_loaded = self
+                .runtime_adapter
+                .load_mem_trie_on_catchup(&shard_uid, &chunk.prev_state_root())?;
+            // If memtrie is enabled but it was not loaded now, it would mean it was already loaded.
+            // It should never happen during state sync.
+            debug_assert!(
```

Comment:
Could we move this debug assert to within the `load_mem_trie_on_catchup` function? This way, we would not have to (1) return `memtrie_loaded` and (2) not expose the trie config.